pros/src/async_runtime/executor.rs

```diff
@@ -79,17 +80,13 @@ impl Executor {
                 if let Poll::Ready(output) = Pin::new(&mut task).poll(&mut cx) {
                     return output;
                 }
+                self.tick();
+                // there might be another future to poll, so we continue without sleeping
+                continue;
```

Comment:
I might be missing something, but in the next iteration of this loop, `woken` will be `false` (as it was `swap`ed on line 79) and this `if` be skipped, meaning control flow ends up right where it would be if there was no continue here.

Reply:
Because we call self.tick, woken might not be false.

Reply:
OK